src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -39,98 +40,234 @@ private ExpressionSyntax ParseInterpolatedStringToken()
+
+                CodeAnalysis.Syntax.InternalSyntax.SyntaxList<InterpolatedStringContentSyntax> result = builder;
+                _pool.Free(builder);
+                content.Free();
+                return result;
+            }
+
+            // Gets the indentation whitespace from the last line of a multi-line raw literal.
+            ReadOnlySpan<char> getIndentationWhitespace(ReadOnlySpan<char> originalTextSpan)
+            {
+                // The content we want to create text token out of.  Effectively, what is in the text sections
+                // minus leading whitespace.
+                var closeQuoteText = originalTextSpan[closeQuoteRange];
+
+                // A multi-line raw interpolation without errors always ends with a new-line, some number of spaces, and
+                // the quotes. So it's safe to just pull off the first two characters here to find where the
+                // newline-ends.
+                var afterNewLine = SlidingTextWindow.GetNewLineWidth(closeQuoteText[0], closeQuoteText[1]);
+                var afterWhitespace = SkipWhitespace(closeQuoteText, afterNewLine);
+
+                Debug.Assert(closeQuoteText[afterWhitespace] == '"');
+                return closeQuoteText[afterNewLine..afterWhitespace];
+            }
+
+            InterpolatedStringContentSyntax? makeContent(
+                ReadOnlySpan<char> indentationWhitespace, StringBuilder content, bool first, bool last, ReadOnlySpan<char> text)
+            {
+                if (text.Length == 0)
+                    return null;
+
+                if (!isMultiLineWithoutErrors)
+                    return SyntaxFactory.InterpolatedStringText(MakeInterpolatedStringTextToken(kind, text.ToString()));
+
+                content.Clear();
+                var currentIndex = 0;
+
+                // If we're not processing the first content chunk, then we must be processing a chunk that came after
+                // an interpolation.  In that case, we need to consume up through the next newline of that chunk as
+                // content that is not subject to dedentation.
+                if (!first)
+                    currentIndex = ConsumeRemainingContentOnLine(content, text, currentIndex);
+
+                // We're either the first item, or we consumed up through a newline from the previous line. We're
+                // definitely at the start of a newline (or at the end).  Regardless, we want to consume each successive
+                // line, making sure its indentation is correct.
+
+                SyntaxDiagnosticInfo? indentationError = null;
+                while (currentIndex < text.Length)
                 {
-                    for (int i = 0; i < interpolations.Count; i++)
+                    var lineStartPosition = currentIndex;
+
+                    // Only bother reporting a single indentation error on a text chunk.
+                    if (error == null)
```

Comment:
Feels like we should still do `SkipWhitespace` (on line 172) for remaining lines after an indentation error

Reply:
we can't bail out the loop (i'll doc).  this loop is effectively consuming all the "lines" of each chunk of text between the interpolations.